plugins/resource_management/app/controllers/resource_management/project_resources_controller.rb

```diff
@@ -21,7 +23,59 @@ def index
       @nearly_full_resources = @all_resources.where("usage <= approved_quota AND current_quota <= approved_quota AND usage >= 0.8 * approved_quota AND usage > 0").to_a
     end
 
-    def new_request
+    def reduce_quota
+      # Note: this function is used in two ways
+      # first show the reduce quota window
+      # second: reduce the quota
+      
+      if params[:resource] 
+        # reduce the quota
+        value = params[:resource][:current_quota]
+        
+        if value.empty?
+          @project_resource.add_validation_error(:current_quota, "empty value is invalid")
+        else 
+          begin
+            # pre check value
+            if @project_resource.approved_quota < @project_resource.data_type.parse(value) &&
+               @project_resource.current_quota < @project_resource.data_type.parse(value)
+              @project_resource.add_validation_error(:current_quota, "wrong value: because the reduced quota value of #{value} is higher than your current quota")
+            elsif @project_resource.usage > @project_resource.data_type.parse(value)
+              @project_resource.add_validation_error(:current_quota, "wrong value: it is now allowed to reduce the quota below your current usage")
+            elsif @project_resource.approved_quota == @project_resource.data_type.parse(value) &&
+                  @project_resource.current_quota == @project_resource.data_type.parse(value)
+              @project_resource.add_validation_error(:current_quota, "wrong value: because the reduced quota value is the same as your current quota")
+            else
+              @project_resource.approved_quota = @project_resource.data_type.parse(value)
+              @project_resource.current_quota = @project_resource.data_type.parse(value)
+            end
+          rescue ArgumentError => e
+            @project_resource.add_validation_error(:current_quota, 'is invalid: ' + e.message)
+          end
+        end
```

Comment:
This paragraph would be easier to read if you put `@project_resource.data_type.parse(value)` into a variable at the start. (This also applies to `domain_admin#reduce_quota`.)